config/logs.js

```diff
@@ -0,0 +1,114 @@
+const _ = require('lodash');
+
+module.exports = {
+  hidFormatter(level, msg, logObject) {
+    // Clone the logged object to avoid mutating important objects (e.g. request)
+    // in case the code continues executing after being logged.
+    const localLogObject = _.clone(logObject || {});
+
+    // Define our meta with some defaults.
+    const metadata = {};
+    metadata.level = level;
+
+    // Log IP if we have it.
+    if (logObject.request && logObject.request.info && logObject.request.info.remoteAddress) {
+      metadata.ip = localLogObject.request.info.remoteAddress;
+    }
+    if (logObject.request && logObject.request.headers && logObject.request.headers['x-forwarded-for']) {
+      metadata.ip = localLogObject.request.headers['x-forwarded-for'];
+    }
+
+    // Include custom user object from log
+    metadata.user = localLogObject.user || {};
+
+    // Include custom oauth object from log
+    metadata.oauth = localLogObject.oauth || {};
```

Comment:
Same question: is it possible that sensitive information may be contained in the `oauth` object?

Reply:
No, because this is populated based on what we chose to log. We don't log the client secret, which is the only sensitive piece of data regarding an OAuth client.

Reply:
I suppose we could explicitly forbid it in the formatter, but this object is created by the developer in each logging instance. It's not an object pulled from the DB.